### PR TITLE
Issue 5639: Added support for BC7 textures on HTML5

### LINKED ISSUE #5639: Basis Universal: BPTC (BC7) for WebGL
**Is your feature request related to a problem? Please describe (REQUIRED):**
RGBA encoded into BC3 hasn't the best quality. Luckily, Chrome and Firefox support BPTC (BC7) -
- https://bugs.chromium.org/p/chromium/issues/detail?id=1013369
- https://developer.mozilla.org/en-US/docs/Web/API/EXT_texture_compression_bptc

**Describe the solution you'd like (REQUIRED):**
Defold can check the [`EXT_texture_compression_bptc`](https://www.khronos.org/registry/webgl/extensions/EXT_texture_compression_bptc/) extension and use it in HTML5 builds.

### PR DESCRIPTION
**Pull request by JCash**
Removed fallback to BC3 since the quality was subpar

### CODE REVIEW

File: engine/graphics/src/opengl/graphics_opengl.cpp [hunk 1]
```diff
@@ -860,9 +860,9 @@ static uintptr_t GetExtProcAddress(const char* name, const char* extension_name,
         if (params->m_PrintDeviceInfo)
         {
             dmLogInfo("Device: OpenGL");
-            dmLogInfo("Renderer: %s\n", (char *) glGetString(GL_RENDERER));
-            dmLogInfo("Version: %s\n", (char *) glGetString(GL_VERSION));
-            dmLogInfo("Vendor: %s\n", (char *) glGetString(GL_VENDOR));
+            dmLogInfo("Renderer: %s", (char *) glGetString(GL_RENDERER));
+            dmLogInfo("Version: %s", (char *) glGetString(GL_VERSION));
+            dmLogInfo("Vendor: %s", (char *) glGetString(GL_VENDOR));
```
Comment (JCash) [inline engine/graphics/src/opengl/graphics_opengl.cpp:865]: Removed unwanted printouts ("\<Empty String\>") in the console

File: engine/graphics/src/graphics.cpp [hunk 2]
```diff
@@ -252,7 +252,6 @@ namespace dmGraphics
         if (IsFormatRGBA(format))
         {
             TEST_AND_RETURN(dmGraphics::TEXTURE_FORMAT_RGBA_BC7);
-            TEST_AND_RETURN(dmGraphics::TEXTURE_FORMAT_RGBA_BC3);
             TEST_AND_RETURN(dmGraphics::TEXTURE_FORMAT_RGBA_ASTC_4x4);
```
Comment (JCash) [inline engine/graphics/src/graphics.cpp:255]: As shown in the issue, the look with 1 bit alpha was not good at all.

File: engine/graphics/src/transcoder/graphics_transcoder_uastc.cpp [hunk 3]
```diff
@@ -152,7 +152,7 @@ namespace dmGraphics
         }
 
 #if defined(TEX_TRANSCODE_DEBUG)
-        dmLogInfo("MAWE: Transcoding: %p from %d to %d (%s -> %s)", path, format, transcoder_format, ToString(format), ToString(transcoder_format));
+        dmLogInfo("Transcoding: %p from %d to %d (%s -> %s)", path, format, transcoder_format, ToString(format), ToString(transcoder_format));
```
Comment (JCash) [inline engine/graphics/src/transcoder/graphics_transcoder_uastc.cpp:155]: Cleanup

File: engine/graphics/src/opengl/graphics_opengl.cpp [hunk 4]
```diff
@@ -956,7 +956,9 @@ static uintptr_t GetExtProcAddress(const char* name, const char* extension_name,
         }
 
         // https://www.khronos.org/registry/OpenGL/extensions/ARB/ARB_texture_compression_bptc.txt
-        if (IsExtensionSupported("GL_ARB_texture_compression_bptc", extensions))
+        if (IsExtensionSupported("GL_ARB_texture_compression_bptc", extensions) ||
+            IsExtensionSupported("GL_EXT_texture_compression_bptc", extensions) ||
+            IsExtensionSupported("EXT_texture_compression_bptc", extensions) )
```
Comment (JCash) [inline engine/graphics/src/opengl/graphics_opengl.cpp:961]: The BC7 support

**Review by britzl — COMMENTED** [review]
Looks good!

**Pull request merged**